File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i, j = 0, len(numbers)-1
        while True:
            sum = numbers[i] + numbers[j]
            if sum < target:
                i += 1
            elif (sum > target):
                j -= 1
            else:
                return ([i+1, j+1])

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
    let l = 0;
    let r = numbers.length - 1;

    // since it's guaranteed that there is only one solution
    // we can use while true
    let num;
    while (true) {
        num = numbers[l] + numbers[r];
        if (num === target) return [l + 1, r + 1];
        if (num > target) r -= 1;
        if (num < target) l += 1;
    }
};